R/importM.COSMIC.R

```diff
@@ -0,0 +1,348 @@
+                                 col_types = readMask,
+                                 n_max = Inf)
+
+    if (!silent) {
+        cat("Done reading from ", fNameCNV, ".\n")
+    }
+
+    # ==== MAP COSMIC IDs TO HGNC GENE SYMBOLS ================================
+
+    ## Used the following call to generate fastMapCosmic.rds:
+    ## fastMapGenerate("hgnc_complete_set.txt", "symbol",
+    ##                 "cosmic", type = "Cosmic",
+    ##                 outputName = "../inst/extdata/fastMapCosmic.rds")
+
+    # load fastMap hash table
+    fastMapCosmic <- readRDS(system.file("extdata",
+                                       "fastMapCosmic.rds",
+                                       package = "rete"))
+
+    # Validate that the type of the hash table is "Cosmic"
+    if (!is.null(attributes(fastMapCosmic)$type)) {
+        if ("Cosmic" != attributes(fastMapCosmic)$type) {
+            stop(sprintf("Expected hash table type: Cosmic.\nSupplied hash table type: %s",
+                         attributes(fastMapCosmic)$type))
+        }
+    } else {
+        stop("Supplied hash table does not have a type attribute.")
+    }
+
+
+    # ==== CONSTRUCT rCNA OBJECT ==============================================
+
+    if (!silent) {
+        cat("Generating rCNA object using CNA data in ", fNameCNV, ".\n")
+    }
+
+    barcodes <- c()
+    hgncSym <- c()
+
+    for (i in 1:nrow(dataCNV)) {
+
+        # Generate all barcodes for the mutations in dataCNV
+        barcodes <- append(barcodes,
+                           paste("TCGA-",
+                                 dataCNV[i,]$ID_SAMPLE, "-",
+                                 dataCNV[i,]$ID_TUMOUR,
+                                 sep = ""))
+
+        # Generate all HGNC symbols using fastMap()
+        hgncSym <- append(hgncSym,
+                          fastMap(dataCNV[i,]$ID_GENE,
+                                  fastMapCosmic,
+                                  type = "Cosmic",
+                                  dev = TRUE))
+
+    }
+
+    # Create a matrix for storing CNV counts.
+    #     nrow is equal to number of unique HGNC symbols
+    #     ncol is equal to the number of unique barcodes.
+    rCNA <- matrix(nrow = length(unique(hgncSym)),
+                   ncol = length(unique(barcodes)))
+
+    # Set the row and column names of rCNA to the
+    # unique HGNC symbols and unique barcodes
+    rownames(rCNA) <- unique(hgncSym)
+    colnames(rCNA) <- unique(barcodes)
+
+    # Populate rCNA matrix with CNA counts
+    for (i in 1:nrow(dataCNV)) {
+        rCNA[fastMap(dataCNV[i,]$ID_GENE,
+                     fastMapCosmic,
+                     type = "Cosmic",
+                     dev = TRUE),
+             paste("TCGA-",
+                   dataCNV[i,]$ID_SAMPLE, "-",
+                   dataCNV[i,]$ID_TUMOUR,
+                   sep = "")] <- as.numeric(dataCNV[i,]$TOTAL_CN)
+    }
+
+    if (!silent) {
+        cat("rCNA object successfully generated.\n")
+    }
+
+
```

Comment:
What kind of metadata are you looking for? I've included:

if (!silent) {
        cat("rCNA object successfully generated.\n")
        cat(sprintf("  Number of unqie samples: %d\n", ncol(rCNA)))
        cat(sprintf("  Number of unique genes: %d\n", nrow(rCNA)))
        cat(sprintf("  Total number of CNAs: %d\n", sum(as.numeric(dataCNV$TOTAL_CN))))
    }

...anything else?